bin/api-docs/gen-components-docs/markdown/props.js

```diff
@@ -0,0 +1,53 @@
+function renderPropType( type ) {
+	switch ( type.name ) {
+		case 'enum': {
+			const MAX_ENUM_VALUES = 10;
+			const string = type.value
+				.slice( 0, MAX_ENUM_VALUES )
+				.map( ( { value } ) => value )
+				.join( ' | ' );
+
+			if ( type.value.length > MAX_ENUM_VALUES ) {
+				return `${ string } | ...`;
+			}
+			return string;
+		}
+		default:
+			return type.name;
+	}
+}
+
+function generateMarkdownPropsJson( props, { headingLevel = 2 } = {} ) {
+	const sortedKeys = Object.keys( props ).sort( ( [ a ], [ b ] ) =>
+		a.localeCompare( b )
+	);
+
+	const propsJson = sortedKeys
+		.flatMap( ( key ) => {
+			const prop = props[ key ];
+
+			if ( prop.description.includes( '@ignore' ) ) {
+				return null;
+			}
+
+			return [
+				{ [ `h${ headingLevel + 1 }` ]: `\`${ key }\`` },
+				prop.description,
+				{
+					ul: [
+						`Type: \`${ renderPropType( prop.type ) }\``,
```

Comment:
For displaying the prop type, I opted to use the <code>- Type: `fooType`</code> format rather than the <code>### propName: `fooType`</code> format, because it is more adaptable in cases where the type is long (like an enum).

Reply:
Sounds sensible to me, especially as the first iteration.

We could later consider more advanced layouts (like maybe a table layout inspired from Storybook), but that's a story for another day.